packages/api-reference/src/components/ApiReferenceLayout.vue

```diff
@@ -257,6 +258,28 @@ provide(INTEGRATION_SYMBOL, () =>
     : 'vue',
 )
 
+// ---------------------------------------------------------------------------/
+// HANDLE MAPPING CONFIGURATION TO INTERNAL REFERENCE STATE
+
+/** Helper utility to map configuration props to the ApiReference internal state */
+function mapConfigToState<K extends keyof ReferenceConfiguration>(
```

Comment:
copy pasted for now but it will get deleted in the other component in my other branch